<launch>
  <!-- Include utilities from xarm -->
  <rosparam command="load" file="$(find xarmrob_util)/src/robot_xarm_info.yaml" />

  <arg name="model" default="$(find xarmrob_util)/src/urdf/robot-xarm.urdf" />
  <arg name="gui" default="false" />
  <arg name="rvizconfig" default="$(find xarmrob_util)/src/urdf/urdf.rviz" />

  <param name="robot_description" command="xacro --inorder $(arg model)" />
  <param name="use_gui" value="$(arg gui)"/>

  <!-- RVIZ preview -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <group if="$(arg var)">
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
    <node name="draw_pen_rviz" pkg="xarmrob" type="draw_pen_rviz.py" />
  </group>
  
  <node pkg="xarmrob" name="forward_kinematics" type="forward_kinematics.py" />

  <!-- Utility node to convert angles to microsecond commands and send them to the arm. -->
  <node pkg="xarmrob" name="command_xarm" type="command_xarm.py" />



  <!-- This is the target SVG to draw -->
  <arg name="target_svg" default="smiley.svg" />
  <param name="target_svg" value="$(find jerry_draw)/src/$(arg target_svg)" />

  <!-- Load parameters for robot_xarm and jerry_draw -->
  <rosparam command="load" file="$(find xarmrob_util)/src/robot_xarm_info.yaml" />
  <rosparam command="load" file="$(find jerry_draw)/src/jerry_info.yaml" />

  <!-- Grab the camera projection matrixes -->
  <param name="cam_matrix" value="$(find jerry_draw)/src/cam_matrix.p" />
  <param name="dist_matrix" value="$(find jerry_draw)/src/dist_matrix.p" />

  <!-- Launch nodes -->
  <node name="get_marker_location" pkg="jerry_draw" type="get_marker_location.py" />
  <node name="inverse_kinematics" pkg="jerry_draw" type="inverse_kinematics.py" />
  <node name="draw_svg" pkg="jerry_draw" type="draw.py" args="--svg" />
    
</launch>